### PR TITLE
[Refactor] Video grid

### PR DESCRIPTION
**Pull request by johnxnguyen**
### Issues

Unfortunately the current video grid implementation does not support sorted datasources.

### Causes

The datasource of the `GridView` (`videoStreamViews`) is a collection of `UIView` which contains no domain information. Additionally, we are not able to provide a sorted datasource because `videoStreamViews`  is manipulated through the `append(view: UIView)` and `remove(view: UIView)` methods. This means that all new streams are appended to end of the list when they may need to be inserted at a particular position.

### Solutions

1. Convert `GridView` to a `UICollectionView`.
2. Delegate datasource responsibilities to its owner `VideoGridViewController`
3. Use `DifferenceKit` to update the `GridView`, avoiding complicated collection manipulation and preserving the ordering of items.
### Notes

Some other minor refactors were made where possible.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Calling/VideoGridView/GridView.swift [hunk 1]
```diff
@@ -18,86 +18,77 @@
...
         case column
+
     }
-    
-    private enum ParticipantAmount {
+
+    enum ParticipantAmount {
+
         case moreThanTwo
         case twoAndLess
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/Calling/VideoGridView/GridView.swift:79]: should it rename to `twoOrLess` ?

File: Wire-iOS/Sources/UserInterface/Calling/VideoGridView/GridCell.swift [hunk 2]
```diff
@@ -26,6 +26,7 @@ class GridCell: UICollectionViewCell {
         guard !contentView.subviews.contains(streamView) else { return }
         contentView.subviews.forEach { $0.removeFromSuperview() }
         contentView.addSubview(streamView)
+        streamView.translatesAutoresizingMaskIntoConstraints = false
         streamView.fitInSuperview()
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Calling/VideoGridView/GridCell.swift:30]: I think that `fitInSuperview` is deprecated

File: Wire-iOS/Sources/UserInterface/Calling/VideoGridView/GridView.swift [hunk 3]
```diff
@@ -18,86 +18,77 @@
...
-        }
-    }
-    
+/// A collection view that displays its items in a dynamic grid layout
+/// depending on the number of items.
+///
+/// In a vertical orientation the grid generally follows a 2 columns x n rows layout.
+/// There are two special cases: firstly a single item will occupy the entire grid,
+/// and secondly two items will form a 1 column x 2 rows layout. In a horizontal
+/// orientation, columns and rows and swapped.
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Calling/VideoGridView/GridView.swift:27]:
```suggestion
/// orientation, columns and rows are swapped.
```

File: Wire-iOS/Sources/UserInterface/Calling/VideoGridView/GridView.swift (hunk 1 above)
Comment (johnxnguyen) [reply Wire-iOS/Sources/UserInterface/Calling/VideoGridView/GridView.swift]: Done

**Pull request merged**